packages/interface-connection-manager/src/index.ts

```diff
@@ -24,6 +24,14 @@ export interface ConnectionManager extends EventEmitter<ConnectionManagerEvents>
    * Close our connections to a peer
    */
   closeConnections: (peer: PeerId) => Promise<void>
+
+  /**
+   * Invoked after an incoming connection is opened but before PeerIds are
+   * exchanged, this lets the ConnectionManager check we have sufficient
+   * resources to accept the connection, otherwise it will be closed and an
+   * error thrown
+   */
+  acceptIncomingConnection: (maConn: MultiaddrConnection) => Promise<void>
```

Comment:
shouldn't this return a boolean or are we expecting implementation to throw?

Reply:
The implementation throws.  It could accept a boolean but throwing lets the error be specific about the failure, though we could just log it instead?

Reply:
I think logging errors and returning boolean would make more sense as you cannot really handle that error up the stack other than logging it. On other hand, it's gonna be obvious to the implementer what it needs to do.